### DISCUSSION TITLE
Add trash support for macOS

### DISCUSSION THREAD

**Discussion by p-mng** [jarun/nnn#1963] [Ideas]
It seems like macOS 14 ("Sonoma", introduced in late 2023) introduced a built-in `trash` command (see `man 8 trash`). Currently, nnn only supports `trash-put` and `gio trash`. I'm aware of #1779, but neither [trash-cli](https://github.com/andreafrancia/trash-cli/issues/323) nor [gio trash](https://gitlab.gnome.org/GNOME/glib/-/issues/1161) include macOS support.

I was wondering if you would be willing to add support for this command, for example using `NNN_TRASH=3`?

man page for reference:

```
TRASH(8)                           System Manager's Manual                           TRASH(8)

NAME
     trash – Moves files and directories to the user trash folder

SYNOPSIS
     trash [-h] [--help] [-s] [-stopOnError] [-v] [--verbose] FILE [FILE...]

DESCRIPTION
     The trash moves files and directories into the user trash folder.  The options are as
     follows:

     -h | --help
                display usage information for the tool and exit

     -v | --verbose
                display more verbose status

     -s | --stopOnError
                exit with an error if any move to a trash folder fails
```

**Comment by N-R-K** [comment]
I can't find the issue/discussion, but there was a proposal to make `NNN_TRASH=3` call out to a custom plugin (e.g `~/.config/nnn/plugins/.trash`) so that the user can customize that plugin to use whatever trashing method they prefer.

Feel free to raise a PR.

**Comment by p-mng** [comment]
> make `NNN_TRASH=3` call out to a custom plugin (e.g `~/.config/nnn/plugins/.trash`)

I don't feel familiar enough with the project (and C in general) to add this feature, but I added support for the `trash` command (using `NNN_TRASH=3`) in my fork at <https://github.com/p-mng/nnn/commit/cef572c0fde839e7e3dba8c3a2c4ba42596eb962>. If you would be willing to add this feature and take a look at the changes I can open a PR.

**Comment by N-R-K** [comment]
@p-mng You can now set `$NNN_TRASH` to a custom command and use whatever trashing utility you prefer. See the [wiki](https://github.com/jarun/nnn/wiki/Usage#configuration) and manpage for more info.

**Comment by p-mng** [comment]
Great, thanks!
